src/statistics.jl

```diff
@@ -409,6 +419,21 @@ end
     kwargs...,
 )
 
+function Statistics.mean!(
+    M::Manifold,
```

Comment:
```suggestion
    M:: AbstractEmbeddedManifold,
```
I think this only works for embedded manifolds, so we have to take that into account for this type.

Reply:
I think we also want it to work for wrapped embedded manifolds (when the outermost thing is not `<:AbstractEmbeddedManifold`).

Reply:
should we then maybe restrict it to abstract decorators? Or do you think that is too restrictive?

Reply:
There doesn't seem to be any good reason for restricting it, the primary reason for restricting argument types should be polymorphic dispatch: https://white.ucc.asn.au/2020/04/19/Julia-Antipatterns.html and here there are no methods we could potentially be concerned about.

Reply:
ok, then let's leave this as is.